SofaKernel/framework/sofa/core/objectmodel/BaseObjectDescription.h

```diff
@@ -66,7 +66,7 @@ class SOFA_CORE_API BaseObjectDescription
 
     typedef std::map<std::string,Attribute> AttributeMap;
 
-    BaseObjectDescription(const char* name=nullptr, const char* type=nullptr);
+    BaseObjectDescription(const std::string name=nullptr, const std::string type=nullptr);
```

Comment:
replace with `const std::string name=""` or `const std::string& name=""`
Underlying code likely has to be updated, I assume that it rely on `if(x == nullptr)` constructs.